File: codeforces/1869/B.py
def find_route(a, b):
    dx = max(a[0] - b[0], b[0] - a[0])
    dy = max(a[1] - b[1], b[1] - a[1])
    return dx + dy

ts = int(input())

for _ in range(ts):
    n, k, a, b = list(map(int, input().split()))
    a -= 1
    b -= 1

    cities = []
    x = [float('inf'), float('-inf')]
    y = [float('inf'), float('-inf')]
    np1 = float('inf')
    np2 = float('inf')

    for i in range(n):
        cities.append(list(map(int, input().split())))

    for i in range(k):
        tgv = find_route(cities[i], cities[a])
        np1 = min(np1, tgv )
        
        tgv = find_route(cities[i], cities[b])
        np2 = min(np2, tgv)

        x[0] = min(x[0], cities[i][0])
        x[1] = max(x[1], cities[i][0])
        y[0] = min(x[0], cities[i][1])
        y[1] = max(x[0], cities[i][1])
    
    a, b = cities[a], cities[b]
    directx = max(a[0] - b[0], b[0] - a[0])
    directy = max(a[1] - b[1], b[1] - a[1])
    direct = directx + directy
        
    ind = np1 + np2

    print(min(ind, direct))

File: codeforces/1869/B.c
#include <stdio.h>
#include <limits.h>

long find_route(long a[2], long b[2]) {
    long dx = a[0] > b[0] ? a[0] - b[0] : b[0] - a[0];
    long dy = a[1] > b[1] ? a[1] - b[1] : b[1] - a[1];
    return dx + dy;
}

int main() {
    freeopen()
    int ts;
    scanf("%d", &ts);

    while (ts--) {
        int n, k;
        long a, b;
        scanf("%d %d %ld %ld", &n, &k, &a, &b);
        a--;
        b--;

        long cities[n][2];
        long x[2] = {LONG_MAX, LONG_MIN};
        long y[2] = {LONG_MAX, LONG_MIN};
        long np1 = LONG_MAX;
        long np2 = LONG_MAX;

        for (int i = 0; i < n; i++) {
            scanf("%ld %ld", &cities[i][0], &cities[i][1]);
        }

        for (int i = 0; i < k; i++) {
            long tgv = find_route(cities[i], cities[a]);
            np1 = tgv < np1 ? tgv : np1;

            tgv = find_route(cities[i], cities[b]);
            np2 = tgv < np2 ? tgv : np2;

            x[0] = cities[i][0] < x[0] ? cities[i][0] : x[0];
            x[1] = cities[i][0] > x[1] ? cities[i][0] : x[1];
            y[0] = cities[i][1] < y[0] ? cities[i][1] : y[0];
            y[1] = cities[i][1] > y[1] ? cities[i][1] : y[1];
        }

        long directx = a > b ? a - b : b - a;
        long directy = a > b ? a - b : b - a;
        long direct = directx + directy;

        long ind = np1 + np2;

        printf(">>> %ld\n", ind < direct ? ind : direct);
    }

    return 0;
}
